File: cell_element_current_sum.py
import numpy as np

def cell_element_current_sum(node, iv, ims, t):
    node = np.asarray(node)
    isum = ims
    
    for i in range(1, 6):              # i từ 1 đến 5 giống MATLAB
        idx = i - 1                    # chuyển sang index Python 0-4
        if node[idx] > 0:
            element_idx = int(node[idx]) - 1   # giả sử phần tử trong iv bắt đầu từ 1
            vt = iv[element_idx]               # ma trận dòng của phần tử này
            Nxi = vt.shape[1]                  # số cột (1 hoặc 2)
            
            if (i % 2 == 0) or (i == 5):
                # Nhánh chẵn hoặc nhánh 5 → dòng chảy RA khỏi nút → lấy cột đầu, trừ đi
                n = 0          # Python index bắt đầu từ 0
                k = -1.0
            else:
                # Nhánh lẻ → dòng chảy VÀO nút → lấy cột cuối, cộng vào
                n = Nxi - 1
                k = 1.0
                
            isum += k * vt[t, n]
            
    return isum

File: origin/cell_element_current_sum.m
function[isum]=cell_element_current_sum(node,iv,ims,t)
%node = matrix(1,5) elements need calculating sum in matrix x


isum = ims;
for i = 1:5,
    if node(i)>0,
        vt = iv{node(i)};
        Nxi = length(vt(t,:));
        if (mod(i,2)==0)|(i==5),
            n = 1;
            k = -1;
        else
            n = Nxi;
            k = 1;
        end
        isum = isum+k*vt(t,n);        
    end
end

